common/dns.go

```diff
@@ -0,0 +1,65 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/wait"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func runDNS() {
+	Context("dns", func() {
+		var (
+			resolver *net.Resolver
+			values   *application.DefaultAppsValues
+		)
+
+		BeforeEach(func() {
+			values = &application.DefaultAppsValues{}
+			err := Framework.MC().GetHelmValues(Cluster.Name, Cluster.Namespace, values)
+			Expect(err).NotTo(HaveOccurred())
+
+			resolver = &net.Resolver{
+				PreferGo:     true,
+				StrictErrors: true,
+				Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+					d := net.Dialer{
+						Timeout: time.Millisecond * time.Duration(10000),
+					}
+					return d.DialContext(ctx, "udp", "8.8.4.4:53")
+				},
+			}
+		})
+
+		It("sets up the api DNS records", func() {
+			apiDomain := fmt.Sprintf("api.%s", values.BaseDomain)
+			var records []net.IP
+			Eventually(func() error {
+				var err error
+				records, err = resolver.LookupIP(context.Background(), "ip", apiDomain)
+				return err
```

Comment:
I think we could do with some logging in here to show that something is still happening. The logs make it look like this test just got stuck.